redisinsight/ui/src/pages/vector-search/query/HeaderActions.tsx

```diff
@@ -1,24 +1,36 @@
-import React, { useState } from 'react'
+import React from 'react'
 import { StyledHeaderAction, StyledTextButton } from './HeaderActions.styles'
 import { ManageIndexesDrawer } from '../manage-indexes/ManageIndexesDrawer'
 
-export const HeaderActions = () => {
-  const [isManageIndexesDrawerOpen, setIsManageIndexesDrawerOpen] =
-    useState<boolean>(false)
+type HeaderActionsProps = {
+  isManageIndexesDrawerOpen: boolean
+  setIsManageIndexesDrawerOpen: (value: boolean) => void
+  isSavedQueriesOpen: boolean
+  setIsSavedQueriesOpen: (value: boolean) => void
+}
 
-  return (
-    <>
-      <StyledHeaderAction data-testid="vector-search-header-actions">
-        <StyledTextButton>Saved queries</StyledTextButton>
-        <StyledTextButton onClick={() => setIsManageIndexesDrawerOpen(true)}>
-          Manage indexes
-        </StyledTextButton>
-      </StyledHeaderAction>
+export const HeaderActions = ({
+  isManageIndexesDrawerOpen,
+  setIsManageIndexesDrawerOpen,
+  isSavedQueriesOpen,
+  setIsSavedQueriesOpen,
+}: HeaderActionsProps) => (
+  <>
+    <StyledHeaderAction data-testid="vector-search-header-actions">
+      <StyledTextButton
+        variant="primary"
+        onClick={() => setIsSavedQueriesOpen(!isSavedQueriesOpen)}
+      >
+        Saved queries
+      </StyledTextButton>
+      <StyledTextButton onClick={() => setIsManageIndexesDrawerOpen(true)}>
```

Comment:
There is something wrong with this state, for some reason, we can no longer open the drawer.

https://github.com/user-attachments/assets/ae33b0f7-1a2a-42e3-8475-420fbc4435bb